backend/monitoring_integrations/datadog_adapter.py

```
@@ -0,0 +1,39 @@
+                })
+            elif "events" in payload and isinstance(payload["events"], list):
+                for e in payload["events"]:
+                    alerts.append({
+                        "provider": "datadog",
+                        "component": e.get("title", "datadog_event"),
+                        "severity": e.get("alert_type", e.get("event_type", "info")),
+                        "message": e.get("text", e.get("body", "")),
+                        "metadata": e,
+                    })
+            else:
+                raise AdapterParseError("Unrecognized Datadog payload")
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)